HotFix/Core/FIXMessageWriter.cs

```diff
@@ -8,123 +8,159 @@ public class FIXMessageWriter
         private const byte SOH = (byte)'\u0001';
         private const byte EQL = (byte) '=';
 
-        internal readonly byte[] _buffer;
-        internal int _end;
-
-        private readonly int _headerEnd;
+        private readonly byte[] _body;
         private int _bodyEnd;
-        private int _trailerEnd;
 
+        internal readonly byte[] Buffer;
+        internal int End;
 
-        public FIXMessageWriter(int maxLength, string beginString)
+        public FIXMessageWriter(int maxLength)
         {
-            _buffer = new byte[maxLength];
-
-            _end += _buffer.WriteString(_end, "8=");
-            _end += _buffer.WriteString(_end, beginString);
-            _buffer[_end++] = SOH;
-
-            _end += _buffer.WriteString(_end, "9=");
-            _end += _buffer.WriteString(_end, "00000");
-            _buffer[_end] = SOH;
-
-            _headerEnd = _end;
-            _end++;
-
-            _end += _buffer.WriteString(_end, "35=");
+            Buffer = new byte[maxLength];
+            _body = new byte[maxLength];
         }
 
-        public void Prepare(string messageType)
+        public FIXMessageWriter Clear()
         {
-            _end = _headerEnd + 4;
+            End = 0;
+            _bodyEnd = 0;
 
-            _end += _buffer.WriteString(_end, messageType);
-            _buffer[_end++] = SOH;
+            return this;
         }
 
-        public void Set(int tag, string value)
+        public FIXMessageWriter Prepare(string beginString, string messageType, long seqNum, DateTime sendingTime, string sender, string target)
         {
-            _end += _buffer.WriteInt(_end, tag);
-            _buffer[_end++] = EQL;
+            // write beginstring
+            End = Buffer.WriteString(0, "8=");
+            End += Buffer.WriteString(End, beginString);
+            Buffer[End++] = SOH;
+
+            // write length placeholder
+            // Should probably measure how many digits are needed to represent maxLength and use that many zeros
```

Comment:
Could you put this comment as either a `NOTE: ` or a `TODO: ` so it's highlighted in VS later on. Also, since you're here, would you mind capitalising your comments etc